components/launcher/src/server/mod.rs

```diff
@@ -509,13 +513,61 @@ fn setup_connection(server: IpcOneShotServer<Vec<u8>>) -> Result<(Receiver, Send
     }
 }
 
+/// Return whether the given version string matches SUP_VERSION_REQ parsed as
+/// a semver::VersionReq.
+///
+/// Example inputs (that is `hab-sup --version` outputs):
+/// hab-sup 0.59.0/20180712161546
+/// hab-sup 0.62.0-dev
+fn is_supported_supervisor_version(version_ouput: String) -> bool {
+    if let Some(version_str) = version_ouput
+        .split(' ')                      // ["hab-sup", <version-number>]
+        .last()                          // drop "hab-sup", keep <version-number>
+        .unwrap_or("")                   //
```

Comment:
I think the `.unwrap_or("")` can be replaced with `.unwrap()`, since `split(' ')` will always return an [`Split` iterator](http://doc.rust-lang.org/1.28.0/alloc/slice/struct.Split.html) with at least on element (even if the input is the empty string).

It would be possible to use map like this:
```rust
    if let Some(version_str) = version_output
        .split(' ')
        .last()
        .map(|s| s.split(|c| c == '/' || c == '-').next().unwrap())
    {
```

But to me the nested `Option` makes things more complex, so I prefer the existing approach.